pkg/controller/rds/dbinstance/setup.go

```diff
@@ -0,0 +1,190 @@
+	for i, v := range cr.Spec.ForProvider.VPCSecurityGroupIDs {
+		obj.VpcSecurityGroupIds[i] = aws.String(v)
+	}
+	obj.DBSecurityGroups = make([]*string, len(cr.Spec.ForProvider.DBSecurityGroups))
+	for i, v := range cr.Spec.ForProvider.DBSecurityGroups {
+		obj.DBSecurityGroups[i] = aws.String(v)
+	}
+	return nil
+}
+
+func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.DBInstance, _ *svcsdk.CreateDBInstanceOutput, _ managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	conn := managed.ConnectionDetails{
+		xpv1.ResourceCredentialsSecretUserKey: []byte(aws.StringValue(cr.Spec.ForProvider.MasterUsername)),
+	}
+	pw, _, err := rds.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, "cannot get password from the given secret")
+	}
+	if pw != "" {
+		conn[xpv1.ResourceCredentialsSecretPasswordKey] = []byte(pw)
+	}
+	return managed.ExternalCreation{
+		ConnectionDetails: conn,
+	}, nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBInstance, obj *svcsdk.DeleteDBInstanceInput) (bool, error) {
+	obj.DBInstanceIdentifier = aws.String(meta.GetExternalName(cr))
+	obj.FinalDBSnapshotIdentifier = aws.String(cr.Spec.ForProvider.FinalDBSnapshotIdentifier)
+	obj.SkipFinalSnapshot = aws.Bool(cr.Spec.ForProvider.SkipFinalSnapshot)
+	return false, nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBInstance, resp *svcsdk.DescribeDBInstancesOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	switch aws.StringValue(resp.DBInstances[0].DBInstanceStatus) {
+	case "available":
+		cr.SetConditions(xpv1.Available())
+	case "deleting", "stopped", "stopping":
+		cr.SetConditions(xpv1.Unavailable())
+	case "creating":
+		cr.SetConditions(xpv1.Creating())
+	}
+	return obs, nil
+}
+
+func filterList(cr *svcapitypes.DBInstance, obj *svcsdk.DescribeDBInstancesOutput) *svcsdk.DescribeDBInstancesOutput {
+	resp := &svcsdk.DescribeDBInstancesOutput{}
+	for _, dbInstance := range obj.DBInstances {
+		if aws.StringValue(dbInstance.DBInstanceIdentifier) == meta.GetExternalName(cr) {
+			resp.DBInstances = append(resp.DBInstances, dbInstance)
+			break
+		}
+	}
+	return resp
+}
+
+func (e *custom) savePasswordSecret(ctx context.Context, cr *svcapitypes.DBInstance, pw string) error {
+	if cr.Spec.ForProvider.MasterUserPasswordSecretRef == nil {
+		return errors.New("no MasterUserPasswordSecretRef given, unable to save password")
+	}
+	ref := cr.Spec.ForProvider.MasterUserPasswordSecretRef
+	nn := types.NamespacedName{
+		Name:      ref.Name,
+		Namespace: ref.Namespace,
+	}
+	sc := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nn.Name,
+			Namespace: nn.Namespace,
+		},
+	}
```

Comment:
```suggestion
	sc := &corev1.Secret{}
```

Metadata here will be filled out by the `Get` call.